contracts/src/Teleporter/TeleporterMessenger.sol

```diff
@@ -285,6 +280,11 @@ contract TeleporterMessenger is ITeleporterMessenger, ReentrancyGuards {
             revert InvalidOriginSenderAddress();
         }
 
+        // If the blockchain ID has yet to be initialized, try to do so now.
+        if (blockchainID == bytes32(0)) {
```

Comment:
Rather than explicitly checking at the callsite, we could modify `initializeBlockchainID` to be a no-op if the `blockchainID` has already been set. I imagine that would save some gas in `receiveCrossChainMessage`, at the cost of making it less obvious that a standalone call to `initializeBlockchainID` had no effect by removing the revert.

Reply:
not the biggest fan of having a no-op call for `initializeBlockchainID`. We could consider have an extra state `isInitialized` that gets set, and check here if it's initialized, and call `initializeBLockchainID` if not. But this approach would cost an extra state variable to store.

Reply:
Updated to go with the lazy initialization pattern, which removed `initializeBlockchainID` all together. 👍